spec/std/string_spec.cr

```diff
@@ -967,6 +968,8 @@ describe "String" do
         it { "日本語日本語".index('本', 2).should eq(4) }
         it { "xyz\xFFxyz".index('\u{FFFD}', 2).should eq(3) }
         it { "xyz\xFFxyz".index('\u{FFFD}', 4).should be_nil }
+        it { "日本\xFF語".index('\u{FFFD}', 2).should eq(2) }
+        it { "日本\xFF語".index('\u{FFFD}', 4).should be_nil }
```

Comment:
`4` would trivially be `nil` because it is equal to the string size:

```suggestion
        it { "日本\xFF語".index('\u{FFFD}', 3).should be_nil }
```